### PR TITLE
Also delete self-affirming or self-infered objects

### PR DESCRIPTION
**Pull request by originalsouth**
### Changes

Also delete self affirming objects

### Issue link

Relevant to but not resolving #3432 (see also #3499)

### Demo
image
Cannot exist anymore

### QA notes
Use https://github.com/originalsouth/VisualOctopoes to see that the graph provided image get garbage collected
Test for any unusual regression in particular for KATFindings.

---

### Code Checklist

- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue.
- [x] I have written unit tests for the changes or fixes I made.
- [x] I have checked the documentation and made changes where necessary.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

- [x] Tickets have been created for newly discovered issues.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: octopoes/octopoes/core/service.py [hunk 1]
```diff
@@ -162,7 +162,7 @@ def get_ooi_tree(
 
     def _delete_ooi(self, reference: Reference, valid_time: datetime) -> None:
         referencing_origins = self.origin_repository.list_origins(valid_time, result=reference)
-        if not referencing_origins:
+        if not [ro for ro in referencing_origins if ro.origin_type != OriginType.AFFIRMATION]:
```
Comment (ammar92) [inline octopoes/octopoes/core/service.py:165]: Nice find. I want to suggest the following to improve this even more:
```suggestion
        if not any(origin for origin in referencing_origins if origin.origin_type != OriginType.AFFIRMATION):
```
- The variable is more descriptive
- Simplified the condition; using this generator expression with `any` would be more Pythonic and efficient than creating a full list to check its emptiness

**Review by ammar92 — COMMENTED** [review]
Nice work. I just have one suggestion to improve this even more. The test looks good and very understandable 👍

**Comment by originalsouth** [comment]
Octopoes is still fundamentally broken, so we'll have more bugs like this. After playing the whack-a-mole game for some hours this resolves some cases like the one #3432.

**Comment by stephanie0x00** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [x] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [x] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
Looks good. When deleting an object all the other objects that are inherited below this object are also removed. Declared objects remain (as is intended behaviour for now). 

### What doesn't work:
n/a

### Bug or feature?:
n/a

**Review by underdarknl — APPROVED** [review]

**Pull request merged**